examples/preprocessing/plot_target_regressor_encoder.py

```diff
@@ -0,0 +1,121 @@
+"""
+=============================
+Target Encoder for Regressors
+=============================
+
+.. currentmodule:: sklearn.preprocessing
+
+The :class:`TargetRegressorEncoder` uses target statistics conditioned on
+the categorical features for encoding. In this example, we will compare
+:class:`TargetRegressorEncoder`, :class:`OrdinalEncoder`, and dropping the
+category on a wine review dataset.
+"""
+
+# %%
+# Loading Data from OpenML
+# ========================
+# First, we load the wine reviews dataset, where the target is the points given
+# be a reviewer:
+from sklearn.datasets import fetch_openml
+
+wine_reviews = fetch_openml(data_id=42074, as_frame=True, parser="liac-arff")
+
+df = wine_reviews.frame
+df.head()
+
+# %%
+# For this example, we use the following subset of numerical and categorical
+# features in the data. The categorical features have a cardinality ranging
+# from 18 to 14810:
+numerical_features = ["price"]
+categorical_features = [
+    "country",
+    "province",
+    "region_1",
+    "region_2",
+    "variety",
+    "winery",
+]
+
+X = df[numerical_features + categorical_features]
+y = df["points"]
+X.nunique().sort_values(ascending=False)
+
+# %%
+# We split the dataset into a training and test set:
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
+
+print(f"Samples in training set: {len(X_train)}\nSamples in test set: {len(X_test)}")
+
+# %%
+# Building and Training Pipelines with Different Encoders
+# =======================================================
+# Dropping the categorical features
+# ---------------------------------
+# As a basline, we construct a pipeline where the categorical features are
+# dropped.
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+prep = ColumnTransformer(
+    [
+        ("num", "passthrough", numerical_features),
+        ("cat", "drop", categorical_features),
+    ]
+)
+
+reg_drop_cats = Pipeline(
```

Comment:
```suggestion
without_categories = Pipeline(
```
Started thinking about the variable name because I tried to work out what the `reg_` meant. Couldn't work it out and then ended up with this suggestion. WDYT? (Shortened things like `cats`, `cnt`, etc are a pet peeve of mine. I prefer the spelt out version as it has nearly no cost but lots of readability improvement.)

Reply:
I'm okay with having better names.

Reply:
Do you like the ones I suggested? Otherwise I will think up some more options that we can choose from/inspire a good name.